contracts/src/LightClient.sol

```diff
@@ -373,6 +384,33 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
         }
     }
 
+    /// @notice update the stateUpdateBlockNumbers and hotShotCommitments arrays each time a new
+    /// finalized state is added
+    /// @dev only maintains maxStateHistoryAllowed non-zero elements in these arrays
+    /// @dev using a FIFO approach to delete elements at the start of the array and storing the new
+    /// initial index in stateHistoryFirstIndex
+    function updateStateHistory(uint256 blockNumber, LightClientState memory state) internal {
+        if (maxStateHistoryAllowed == 0) revert InvalidMaxStateHistory();
+
+        if (stateHistoryCount == maxStateHistoryAllowed) {
+            // the arrays already has the max block states allowed so we clear the first non-empty
+            // element
+            // the arrays are cleared from a FIFO approach
+            delete stateUpdateBlockNumbers[stateHistoryFirstIndex];
+            delete hotShotCommitments[stateHistoryFirstIndex];
+            // and increment the first index so that you know where the first non-zero element is in
+            // the affected arrays
+            stateHistoryFirstIndex++;
+        } else {
+            // increment the num block states recorded, this increment stops when stateHistoryCount
+            // == maxStateHistoryAllowed
+            stateHistoryCount++;
+        }
+        //add the L1 Block to stateUpdateBlockNumbers &  HotShot commitment for the genesis state
+        stateUpdateBlockNumbers.push(blockNumber);
+        hotShotCommitments.push(HotShotCommitment(state.blockHeight, state.blockCommRoot));
```

Comment:
Since we will just push the new elements into the array, why bother deleting the first one?

Reply:
so that the storage is emptied and interacting with the array will consume less gas.

The goal is to maintain a history that isn't larger than maxStateHistoryAllowed. So i delete the first element and increment stateHistoryFirstIndex, so that the history is kept within this bound. The amount if storage required is a predictable and we also get a gas refund with the `delete`.